src/Init/SimpLemmas.lean

```diff
@@ -7,142 +7,126 @@ notation, basic datatypes and type classes
 -/
 prelude
 import Init.Core
+set_option linter.missingDocs true -- keep it documented
 
-@[simp] theorem eq_self (a : α) : (a = a) = True :=
-  propext <| Iff.intro (fun _ => trivial) (fun _ => rfl)
-
-theorem of_eq_true (h : p = True) : p :=
-  h ▸ trivial
+theorem of_eq_true (h : p = True) : p := h ▸ trivial
 
 theorem eq_true (h : p) : p = True :=
```

Comment:
What is the motivation for this kind of change?

Reply:
For the one above, it is at least avoiding the `propext`, which is a bit irrelevant if we consider the fact the theorem is used only by `simp`, and if you use `simp`, you are admitting you have no problem with axioms such as `propext`.

Reply:
I'm not sure what "this kind of change" is referring to here. I don't think I changed any of the proofs in any essential way - they all still have the same statements and use `propext` in all the same places.

Reply:
I was referring to style changes.